pkg/server/module_server.go

```diff
@@ -64,6 +64,7 @@ func newModuleServer(opts Options,
 	reg prometheus.Registerer,
 	promGatherer prometheus.Gatherer,
 	license licensing.Licensing,
+	tracer tracing.Tracer,
```

Comment:
I'd much rather use the OTel `trace.Tracer` interface here, to avoid further proliferation of our own [`tracing.Tracer`](https://github.com/grafana/grafana/blob/main/pkg/infra/tracing/tracing.go#L65) interface, which is largely unnecessary (it dates back to when we supported both OpenTracing/Jaeger and OpenTelemetry, but now we only support the latter).